numpy/core/src/private/ufunc_override.c

```diff
@@ -34,47 +39,33 @@ has_non_default_array_ufunc(PyObject *obj)
 
     /* Fast return for ndarray */
     if ((PyObject *)Py_TYPE(obj) == ndarray) {
-        return 0;
+        return NULL;
     }
     /* does the class define __array_ufunc__? */
-    cls_array_ufunc = PyArray_GetAttrString_SuppressException(
-                          (PyObject *)Py_TYPE(obj), "__array_ufunc__");
+    cls_array_ufunc = PyArray_LookupSpecial(obj, "__array_ufunc__");
     if (cls_array_ufunc == NULL) {
-        return 0;
+        return NULL;
     }
     /* is it different from ndarray.__array_ufunc__? */
-    non_default = (cls_array_ufunc != ndarray_array_ufunc);
+    if (cls_array_ufunc != ndarray_array_ufunc) {
+        return cls_array_ufunc;
+    }
     Py_DECREF(cls_array_ufunc);
-    return non_default;
-}
-
-/*
- * Check whether an object sets __array_ufunc__ = None. The __array_func__
- * attribute must already be known to exist.
- */
-static int
-disables_array_ufunc(PyObject *obj)
-{
-    PyObject *array_ufunc;
-    int disables;
-
-    array_ufunc = PyObject_GetAttrString(obj, "__array_ufunc__");
-    disables = (array_ufunc == Py_None);
```

Comment:
Seemed dumb to look up the attribute again, when we already did given that _"The __array_func__		
 attribute must already be known to exist."_